lib/schedule/minischedule/piece.ex

```diff
@@ -43,15 +44,16 @@ defmodule Schedule.Minischedule.Piece do
     :end
   ]
 
-  @spec hydrate(t(), Schedule.Trip.by_id()) :: t()
-  def hydrate(piece, trips_by_id) do
+  @spec hydrate(t(), Schedule.Trip.by_id(), Timepoint.timepoint_names_by_id()) :: t()
+  def hydrate(piece, trips_by_id, timepoint_names_by_id) do
```

Comment:
To do my last comment, you'll need to hydrate `piece.start.place` and `piece.end.place` in this function.